src/Lean/Meta/Tactic/FunInd.lean

```diff
@@ -0,0 +1,855 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joachim Breitner
+-/
+
+prelude
+import Lean.Meta.Basic
+import Lean.Meta.Match.MatcherApp.Transform
+import Lean.Meta.Check
+import Lean.Meta.Tactic.Cleanup
+import Lean.Meta.Injective -- for elimOptParam
+import Lean.Elab.PreDefinition.WF.Eqns
+import Lean.Elab.PreDefinition.WF.PackMutual
+import Lean.Elab.Command
+
+/-!
+This module contains code to derive, from the definition of a (possibly mutual) recursive function,
+a **functional induction principle** tailored to proofs about that function. For example from:
```

Comment:
As I understand it, this is really relevant for functions defined by well-founded recursion, rather than structural recursion, but this fact isn't state here. Making this front-and-center will also explain the later comment that it expects a `fix` in the sufficiently-normalized definition.

Reply:
Down the road we might extend it to to structurally recursive or possibly even non-recursive functions, but for now you are right.

```suggestion
This module contains code to derive, from the definition of a (possibly mutual) well-founded
recursive function, a **functional induction principle** tailored to proofs about that
function. For example from:
```

Reply:
Now to really split hairs so thin that a microscope can hardly see them: a well-founded recursive function is, in my mind, just a total recursive function. What about "from the definition of a recursive function (or mutually recursive functions) defined by well-founded recursion"?